grin/src/Grin/ExtendedSyntax/Grin.hs

```diff
@@ -54,6 +54,7 @@ _Lit :: Traversal' Val Lit
 _Lit f (Lit l) = Lit <$> f l
 _Lit _ rest    = pure rest
 
+-- NOTE: This will return NM "" for everyrhing that is not a Var
```

Comment:
As far as I know, if it is not a var it leaves the structure untouched. What do you mean by it creates NM "" ?

Reply:
Sorry for the confusion, what I meant is that the way it interacts with `view`, it will always return an empty name for any `Val` besides `Var`.